### ISSUE TITLE
How to fetch the last batch in a multigpu model

### ISSUE THREAD

**Issue by TomorrowIsAnOtherDay** [tensorflow/tensorflow#12924]
The new dataset API is awesome, but I don't know to fetch the last batch in my model 
![image](https://user-images.githubusercontent.com/25046619/30235975-10aad7ce-9543-11e7-9b68-3c25330ee899.png)
![image](https://user-images.githubusercontent.com/25046619/30235993-3d38c9a4-9543-11e7-9698-d13b49e37050.png)

**Comment by cy89** [comment]
@TomorrowIsAnOtherDay I'm sorry, but I'm not understanding from your screenshot what the problem is. You've set a batch size of 12, and the dataset returns 8 batches of that size, consuming 96/100 of the values in your dataset. Are you expecting the batch on the next call to be filled out somehow?

**Comment by TomorrowIsAnOtherDay** [comment]
@cy89 yes, I wanted to feed the last examples(97/98/99)into my multi-gpu model

**Comment by cy89** [comment]
Thanks, @TomorrowIsAnOtherDay. I think the interface is working-as-intended. It's probably a bad idea for datasets to automatically pad your data for you out to the desired batch size, as this might have weird effects on your training. I would expect that randomization plus setting some number of epochs is the answer to uniformly using data that doesn't exactly match the batch size. 

This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by ppwwyyxx** [comment]
I agree it's a bad idea to pad data, but it'll be helpful to return a tensor of smaller batch.
For example, if we want to calculate accuracy on a validation set, each sample needs to be fetched before an OutOfRangeError is raised. In tensorpack we have a [`remainder=False` option](http://tensorpack.readthedocs.io/en/latest/modules/dataflow.html#tensorpack.dataflow.BatchData) for this purpose.

**Comment by TomorrowIsAnOtherDay** [comment]
@cy89 @ppwwyyxx Thanks, Maybe I should provide an parameter to specific whether to drop the last batch or not.
@ppwwyyxx But in my case, The problem is not whether to use a small batch size or not to fetch the last batch.I can't fetch the last batch in multigpu model.If I use only one gpu and the problem doesn't exist, because Dataset API will provide the last batch with a smaller batch.
#12855
